### ISSUE TITLE
Does this work on a Teensy 2.0?

### ISSUE THREAD

**Issue by sofakng** [Harvie/ps2dev#13]
Do you know if this would work on a Teensy 2.0?   ...or does it require the Arduino bootloader/code?

**Comment by Harvie** [comment]
Never used teensy, so give it a try and let me know. It certainly requires Arduino API, such as `digitalWrite(pin, LOW);` and similar... Perhaps if you program Teensy using Arduino IDE, this will be probably part of the Teensy support package for Arduino. If you use different framework, it will probably be missing.

Personaly i would accept modifications to add HAL layer to support other platforms, this should be easy to implement. Just few wrappers and ifdefs around GPIO and interrupt code.

**Comment by cyberluke** [comment]
It works, I use it.

**Comment by Harvie** [comment]
@cyberluke Good to know. Thank you. Did you needed to modify something or do some extra setup?

**Comment by cyberluke** [comment]
No, works out of the box. You just need to download Arduino IDE for Teensy, which is called Teensyduino: https://www.pjrc.com/teensy/teensyduino.html

I shared a code with you in another thread for Teensy 3.2

The only change, if you want to use SPI is this code initialization:
```// Satisfy the IDE, which needs to see the include statment in the ino too.
#ifdef dobogusinclude
#include <spi4teensy3.h>
#endif
#include <SPI.h>
